android/sdl_android/src/main/java/com/smartdevicelink/managers/SdlManager.java

```diff
@@ -584,6 +586,9 @@ public void onTransportEvent(List<TransportRecord> connectedTransports, boolean
 				if (sdlSecList != null && !sdlSecList.isEmpty()) {
 					proxy.setSdlSecurityClassList(sdlSecList);
 				}
+				for (FunctionID functionID: onRPCNotificationListeners.keySet()) {
```

Comment:
Get keySet and set to an instance, check for null and empty set before looping through

Reply:
Okay I updated the pr to do the null checks